### PR TITLE
Remove the dependency on biorxiv to find meca files

### PR DESCRIPTION
**Pull request by E-Kemp**
Closes https://github.com/elifesciences/enhanced-preprints-issues/issues/605

### CODE REVIEW

File: src/activities/parse-docmap.test.ts [hunk 1]
```diff
@@ -5,10 +5,65 @@ const simpleDocmap = `{
...
+        }
+      ],
+      "inputs": []
+    }
+  }
+}`;
+
 describe('parse-docmap-activity', () => {
-  it('parses docmap passed as param', async () => {
+  it('fails to parse a docmap that does not contain a tdmPath', async () => {
```
Comment (scottaubrey) [inline src/activities/parse-docmap.test.ts:58]: parsing could "fail" for lots of reasons - this probably isn't the right description of this failure (there's nothing in simpleDocmap to parse).

File: src/workflows/import-content.ts [hunk 2]
```diff
@@ -18,7 +15,7 @@ const {
 });
 
 export type ImportContentOutput = {
-  preprintPath: string,
+  preprintPath?: string,
```
Comment (scottaubrey) [inline src/workflows/import-content.ts:18]: I would probably just remove this - it was helpful only to report/diagnose an action we're no longer taking (finding the path from biorxiv)

File: src/workflows/import-docmap.ts [hunk 3]
```diff
@@ -22,7 +23,7 @@ export async function importDocmap(url: string): Promise<DocMapImportOutput> {
   const result = await parseDocMap(docMap);
 
   await Promise.all(
-    result.versions.map(async (version, index) => executeChild('importContent', {
+    result.versions.map(async (version, index) => executeChild(importContent, {
```
Comment (scottaubrey) [inline src/workflows/import-docmap.ts:26]: Nice catch and improvement.

File: src/activities/parse-docmap.ts [hunk 4]
```diff
@@ -3,9 +3,15 @@ import { parser } from '@elifesciences/docmap-ts';
 export const parseDocMap = async (docMapInput: string): Promise<parser.ManuscriptData> => {
   const manuscriptData = parser.parsePreprintDocMap(docMapInput);
 
-  if (!manuscriptData) {
+  try {
+    // TO-DO: This is a temporary hack, remove once DocMaps are fixed
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    const tdmPath: string = JSON.parse(docMapInput)['steps']['_:b0']['actions'][0]['outputs'][0]['_tdmPath'];
+    if (manuscriptData?.versions?.[0]?.preprint) manuscriptData.versions[0].preprint.content = tdmPath;
+    else throw Error();
```
Comment (scottaubrey) [inline src/activities/parse-docmap.ts:11]: I'm not sure we want to error here - this is making the assumption the first step has a _tdmPath, and that it should replace version 0's preprint content. The original condition - if manuscriptData is empty we should error - stands as a valid condition to respond to, in additional to the new logic we want.

On the logic itself - I'm not sure it's going to be forward compatible because if the docmap is fixed, we'll replace a valid preprint content URL with an invalid/undefined. 

It also not taking into account revised preprint that isn't step `_:b0` (see step `_:b3` of https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/by-publisher/elife/get-by-manuscript-id?manuscript_id=85111)

Instead, we should probably be looping all steps, all inputs, and constructing an additional content entry to inject into the docmap, then handing the "fixed" docmap to the parser. Happy to explain more if that's not making any sense.

**Review by scottaubrey — CHANGES_REQUESTED** [review]
Thanks @E-Kemp - things are looking mostly good. I've a few tiny nitpick type suggestions, a one bigger requests to consider revised preprint docmaps and forward compatibility. Happy to pair or explain things if you'd like to?

File: src/activities/parse-docmap.test.ts (hunk 1 above)
Comment (E-Kemp) [reply src/activities/parse-docmap.test.ts:58]: I'm thinking something instead like 'fails to parse an invalid docmap'

File: src/S3Bucket.ts [hunk 5]
```diff
@@ -19,6 +19,23 @@ export const getS3Client = () => {
       forcePathStyle: true,
       region: config.s3.region,
     });
+
```
Comment (scottaubrey) [inline src/S3Bucket.ts:22]: I think this is unrelated, and should probably be moved to a separate branch. Also, would it not be easier using [this param](https://aws.amazon.com/blogs/developer/the-aws-sdk-for-javascript-now-supports-amazon-s3-requester-pays-buckets/)?

File: src/activities/parse-docmap.ts [hunk 6]
```diff
@@ -1,11 +1,25 @@
...
+    const contentPaths: string[] = [];
+    Object.keys(steps).forEach((key) => {
+      // Loop through the actions in a step and the corresponding outputs to extract _tdmPaths
+      // eslint-disable-next-line @typescript-eslint/dot-notation
+      steps[key].actions.forEach((action: any) => action.outputs?.forEach((output: any) => { if (output['_tdmPath']) contentPaths.push(output['_tdmPath']); }));
+    });
 
-  return manuscriptData;
+    if (contentPaths.length === manuscriptData?.versions.length) {
+      manuscriptData.versions = manuscriptData.versions.map<VersionedReviewedPreprint>((version, index) => ({ ...version, preprint: { ...version.preprint, content: contentPaths[index] } }));
```
Comment (scottaubrey) [inline src/activities/parse-docmap.ts:18]: There are so many assumptions in here about the structure and makeup of a docmap, but given we invented the _tdmPath I'm guessing it's not going to matter in practice. On that basis, I think it should work OK.

I'm not suggesting doing this, as it's just not valuable enough, but a potential improvement might have been using a Map/object for `contentPaths` to index tdmPaths with their DOIs or the content URL, then hunting for matching DOI/URL in the versions to replace the content URL in versions. This would have at least made sure we're replacing the correct values in the version structure. We then also wouldn't need to error if the number of versions and TDM paths didn't match (which is allowed in a docmap, we just don't do that). 

If [this issue](https://github.com/elifesciences/data-hub-issues/issues/671) is resolved, we'd also need to update this code to look to inputs as well as outputs for TDM paths

**Review by scottaubrey — DISMISSED** [review]
It's a hack. Makes me feel bad, but it works.

Thanks for the work done on it @E-Kemp 👍

File: src/S3Bucket.ts (hunk 5 above)
Comment (E-Kemp) [reply src/S3Bucket.ts:22]: I swear I was googling something like this for ages yesterday and never found it 😭 

But yes I'll move it to a separate branch 👍

File: src/activities/parse-docmap.ts (hunk 6 above)
Comment (E-Kemp) [reply src/activities/parse-docmap.ts:18]: That improvement would make sense but I agree it works _for now_ until _tdmPath is replaced 😂

**Review by scottaubrey — APPROVED** [review]
